tests/helper.py

```diff
@@ -2,6 +2,195 @@
 Helper functions for unit tests.
 """
 
+import unittest
+
+import geomstats.backend as gs
+
+
+class TestGeomstatsMethods(unittest.TestCase):
+    def assertScalar(self, result, n_samples=1):
+        return self.assertTrue(
+            gs.allclose(result.shape, (n_samples, 1)),
+            '\nresult.shape = {} instead of {}'.format(
+                result.shape, (n_samples, 1)))
+
+    def assertVector(self, result, n_samples=1, dim=1):
+        return self.assertTrue(
+            gs.allclose(result.shape, (n_samples, dim)),
+            '\nresult.shape = {} instead of {}'.format(
+                result.shape, (n_samples, dim)))
+
+    def assertAllClose(self, result, expected):
+        result = gs.asarray(result)
+        expected = gs.asarray(expected)
+        return self.assertTrue(
+            gs.allclose(result, expected),
+            '\nresult.shape = {}'
+            '\nexpected.shape = {}'
+            '\nresult = {}'
+            '\nexpected = {}'.format(
+                result.shape,
+                expected.shape,
+                result,
+                expected))
+
+    def check_shape_belongs(self, space):
+        point = space.random_uniform()
+        belongs = space.belongs(point)
+        print('belongs={}'.format(belongs))
+
+        self.assertScalar(belongs)
+
+    def check_shape_belongs_vectorization(self, space, n_samples):
+        n_samples = self.n_samples
+
+        points = self.space.random_uniform(n_samples=n_samples)
+        belongs = self.space.belongs(points)
+
+        self.assertScalar(belongs, n_samples)
+
+    def check_shape_random_uniform(self, space, vec_dim):
+        point = space.random_uniform()
+
+        self.assertVector(point, dim=vec_dim)
+
+    def check_shape_random_uniform_vectorization(self,
+                                                 space,
+                                                 n_samples,
+                                                 vec_dim):
+        point = space.random_uniform(n_samples)
+        self.assertVector(point, n_samples=n_samples, dim=vec_dim)
+
+    def assert_random_uniform_and_belongs(self, space):
+        point = self.space.random_uniform()
+
+        self.assertTrue(self.space.belongs(point))
+
+    def assert_random_uniform_and_belongs_vectorization(self,
+                                                        space,
+                                                        n_samples):
+        n_samples = self.n_samples
+
+        points = self.space.random_uniform(n_samples=n_samples)
+
+        self.assertTrue(gs.all(self.space.belongs(points)))
+
+    def check_shape_exp_vectorization(self, space, n_samples, dim):
```

Comment:
do we still need tests that are so complex now that we've reduced the complexity?